code/modules/mob/new_player/new_player.dm

```diff
@@ -297,22 +291,67 @@
 				to_chat(src, "<span class='notice'>Vote successful.</span>")
 
 
+/mob/new_player/proc/make_me_an_observer()
+	if(!SSticker || SSticker.current_state == GAME_STATE_STARTUP)
+		to_chat(src, "<span class='warning'>The game is still setting up, please try again later.</span>")
+		return FALSE
+
+	if(QDELETED(src) || !src.client)
+		ready = PLAYER_NOT_READY
+		return FALSE
+
+	var/this_is_like_playing_right = alert(src,"Are you sure you wish to observe?\nYou will have to wait at least 5 minutes before being able to respawn!","Player Setup","Yes","No")
```

Comment:
You could combine it and just put it into the if check.

Reply:
done